sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -274,16 +281,89 @@ protected override async ValueTask<Response> UpdateStatusAsync(bool async, Cance
             }
         }
 
-        private static Response GetResponseFromState(OperationState<T> state)
+        private RequestMethod? GetRequestMethod(RehydrationToken? rehydrationToken)
+        {
+            if (rehydrationToken is null)
+            {
+                return null;
+            }
+            var lroDetails = ModelReaderWriter.Write(rehydrationToken, new ModelReaderWriterOptions("J")).ToObjectFromJson<Dictionary<string, string>>();
+            return new RequestMethod(lroDetails["requestMethod"]);
+        }
+
+        private static Response GetResponseFromState(OperationState<T> state, RequestMethod? requestmethod = null)
         {
             if (state.HasSucceeded)
             {
                 return state.RawResponse;
             }
 
+            // if this is a fake delete lro with 404, just return empty response with 200
```

Comment:
I believe in the review we decided to go with 204 here

Reply:
If we take a look at the [fake delete test](https://github.com/Azure/azure-sdk-for-net/blob/bdc47b5799621e4ad09f0236f475646fd5106a14/sdk/resourcemanager/Azure.ResourceManager/tests/Scenario/LroRehydrationTests.cs#L52), the original LRO response returns 200, not 204.

So, should we follow the original status 200 here?

Reply:
updated to 204